pkg/database/gorm.go

```diff
@@ -29,25 +65,22 @@ func NewConnection(config *Config, environment string) (*gorm.DB, error) {
 		connectionString = testDriverName
 	}
 
-	dialector := mysql.New(mysql.Config{
+	return mysql.New(mysql.Config{
 		DSN:        connectionString,
 		DriverName: driverName,
 	})
+}
 
-	db, err := gorm.Open(dialector)
-	if err != nil {
-		return nil, err
-	}
-
-	sqlDB, err := db.DB()
+func databasePoolSettings(gormDB *gorm.DB, config *Config) error {
```

Comment:
I suggest we rename this to `configureDatabase`.

```suggestion
func configureDatabase(gormDB *gorm.DB, config *Config) error {
```

This function is configuring connection settings as well, not just the size of the pool.

I do like that you split this functionality into a separate function.

Reply:
Got it. I don't have a strong opinion – you can leave as-is.